tests/suites/test_suite_psa_crypto.data

```diff
@@ -1433,10 +1433,6 @@ Copy fail: AES, invalid persistent key identifier in attributes
 depends_on:PSA_WANT_ALG_CTR:PSA_WANT_KEY_TYPE_AES:MBEDTLS_PSA_CRYPTO_STORAGE_C
 copy_fail:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:0:PSA_KEY_TYPE_AES:"404142434445464748494a4b4c4d4e4f":PSA_KEY_TYPE_AES:0:PSA_KEY_USAGE_COPY | PSA_KEY_USAGE_ENCRYPT | PSA_KEY_USAGE_EXPORT:PSA_ALG_CTR:0:0:PSA_KEY_LIFETIME_PERSISTENT:PSA_ERROR_INVALID_ARGUMENT
 
-Copy fail: AES, invalid lifetime (unknown location) in attributes
```

Comment:
Just wanted to check we are not losing test coverage here. Is it not possible to have an unknown location when drivers are enabled?

Reply:
This test was targeting the test in psa_crypto_slot_management.c:psa_validate_key_location() that is removed by the commit "psa: Remove MBEDTLS_PSA_CRYPTO_DRIVERS configuration option" in this PR. Without this check, the test still fails but not for the unknown location reason. Another check fails before to reach that point: different location between the source and the destination leading to `psa_copy_key` returning PSA_ERROR_NOT_SUPPORTED and not PSA_ERROR_INVALID_ARGUMENT. The test then triggers the same execution path as "Copy fail: AES, across locations (unsupported) in attributes". To go through the same location test successfully the source and target keys needs to have the same location but then the execution is not going to reach the check of the destination location, it will fail when checking the source location. Thus all together I've chosen to just remove the test case.

Reply:
This sounds wrong. One test case attempts to copy to a valid location which is not the source location, which is currently unsupported but should become supported one day. Another test case attempts to copy to an invalid location, which must never succeed. Even if the test cases currently trigger the same code paths, they are semantically distinct.

I'm also concerned by the removal of `psa_validate_key_location`. This may actually be a preexisting bug. What part of the code is responsible for rejecting invalid locations when there is no driver wrapper (which is the case at least for `psa_key_derivation_output_key`)?

Reply:
Yes I missed this `psa_key_derivation_output_key()` case. There was no check for the key location for this API when MBEDTLS_PSA_CRYPTO_DRIVERS was enabled, but now it is worst, there is no check whatever the configuration.

I think there is a link with #7395. If we say that when there is no driver there should be eventually (not the case currently) no actual driver wrapper functions (just some #define psa_driver_wrapper_xyz mbedtls_psa_xyz) then the check of the key location should not be in wrapper code. One possibility would be that when generating the driver wrapper (when there are some drivers) we generate the list of the external locations. By default (out of the box library) the list would be empty and we would use this list in `psa_validate_key_location()`. Wdyt?